src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -161,31 +168,33 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 }
             }
 
-            if ($oldBuildDir) {
-                $fs->rename($realBuildDir, $oldBuildDir);
-            } else {
-                $fs->remove($realBuildDir);
-            }
             if ($oldCacheDir) {
                 $fs->rename($realCacheDir, $oldCacheDir);
             } else {
                 $fs->remove($realCacheDir);
             }
             $fs->rename($warmupDir, $realCacheDir);
-            // Copy the content of the warmed cache in the build dir
-            $fs->copy($realCacheDir, $realBuildDir);
+
+            if ($useBuildDir) {
+                $fs->rename($realBuildDir, $oldBuildDir);
+                // Copy the content of the warmed cache in the build dir
+                $fs->copy($realCacheDir, $realBuildDir);
```

Comment:
One need to use `mirror()`. 

Also, You just removed `$realCacheDir`, you cannot use it here.